core/src/main/java/com/graphhopper/routing/util/CustomModel.java

```diff
@@ -175,92 +183,46 @@ public static CustomModel merge(CustomModel baseModel, CustomModel queryModel) {
         return mergedCM;
     }
 
-    private static void applyChange(Map<String, Object> mergedSuperMap,
-                                    Object mergedObj, Map.Entry<String, Object> querySuperEntry) {
-        if (mergedObj == null) {
-            // no need for a merge
-            mergedSuperMap.put(querySuperEntry.getKey(), querySuperEntry.getValue());
-            return;
-        }
-        if (!(mergedObj instanceof Map))
-            throw new IllegalArgumentException(querySuperEntry.getKey() + ": entry is not a map: " + mergedObj);
-        Object queryObj = querySuperEntry.getValue();
-        if (!(queryObj instanceof Map))
-            throw new IllegalArgumentException(querySuperEntry.getKey() + ": query entry is not a map: " + queryObj);
+    private interface MergeOp {
```

Comment:
You could add `@FunctionalInterface`

Reply:
It is only used internally so it wouldn't add value.